crates/loader/src/local.rs

```diff
@@ -110,15 +110,12 @@ impl LocalLoader {
         id: &KebabId,
         component: v2::Component,
     ) -> Result<LockedComponent> {
-        outbound_http::allowed_http_hosts::parse_allowed_http_hosts(&component.allowed_http_hosts)?;
-        if let Some(hosts) = &component.allowed_outbound_hosts {
-            spin_outbound_networking::AllowedHosts::parse(hosts)
-                .context("`allowed_outbound_hosts` is malformed")?;
-        }
+        spin_outbound_networking::AllowedHostsConfig::parse(&component.allowed_outbound_hosts)
+            .context("`allowed_outbound_hosts` is malformed")?;
```

Comment:
Definitely low priority but it might be nice to eventually move this parsing into serde for the nice errors you get out of toml deserialization.